dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/helpers/calculate-initial-utilisation-and-fixed-fee.test.ts

```diff
@@ -0,0 +1,176 @@
+import { calculateInitialUtilisation } from '@ukef/dtfs2-common';
+import { calculateInitialUtilisationAndFixedFee, parseDate, hasRequiredValues, RequiredParams } from './calculate-initial-utilisation-and-fixed-fee';
+import { TfmFacilitiesRepo } from '../../../../../repositories/tfm-facilities-repo';
+import { aTfmFacility } from '../../../../../../test-helpers';
+import { calculateInitialFixedFee } from './calculate-initial-fixed-fee';
+
+describe('helpers/calculate-initial-utilisation-and-fixed-fee', () => {
+  describe('parseDate', () => {
+    describe('date passed as null', () => {
+      it('should return an error', () => {
+        expect(() => parseDate(null)).toThrow('Invalid date');
+      });
+    });
+
+    describe('date passed in a correct format', () => {
+      it('should return the formatted date', () => {
+        const date = new Date();
+
+        const result = parseDate(date);
+
+        expect(result).toEqual(date);
+      });
+    });
+
+    describe('date passed as a string', () => {
+      it('should return the formatted date', () => {
+        const date = new Date().toString();
+
+        const result = parseDate(date);
+        const expected = new Date(date);
+
+        expect(result).toEqual(expected);
+      });
+    });
+
+    describe('date passed in EPOCH', () => {
+      it('should return the formatted date', () => {
+        const date = new Date().getTime();
+
+        const result = parseDate(date);
+        const expected = new Date(date);
+
+        expect(result).toEqual(expected);
+      });
+    });
+  });
+
+  describe('hasRequiredValues', () => {
+    const baseParams = {
+      value: 1,
+      interestPercentage: 2,
+      dayCountBasis: 3,
+      coverStartDate: new Date(),
+      coverEndDate: new Date(),
+    } as RequiredParams;
+
+    it('should return true if all values are present', () => {
+      const result = hasRequiredValues(baseParams);
+      expect(result).toEqual(true);
+    });
+
+    it('should return false if the value is not provided', () => {
+      baseParams.value = null;
+
+      const result = hasRequiredValues(baseParams);
+      expect(result).toEqual(false);
+    });
+
+    it('should return false if the interestPercentage is not provided', () => {
+      baseParams.interestPercentage = null;
```

Comment:
Doing this in each test isn't working as expected, e.g in this test, `value` is also null, so this test for "if the interestPercentage is not provided" is actually testing "if value _and_ interestPercentage is not provided"

Need to do this so it only nullifies X field in Y test specifically

```ts
const result = hasRequiredValues({ ...params, dayCountBasis: null });
```

Or this, but I prefer the one liner.

```ts
const params = { ...validParams, dayCountBasis: null };
const result = hasRequiredValues(params);
```